### DISCUSSION TITLE
oracledb with sqlalchemy 1.4 in thick mode in fastapi

### DISCUSSION THREAD

**Discussion by jnre** [sqlalchemy/sqlalchemy#11324] [Usage Questions]
i do want to know if this is possible to setup because i want to be able to have two-phase commit. or is v2.0 possible too?

**Comment by CaselIT** [comment]
Hi

Support was added in v2, but you can follow this guide to use in it in 1.4 https://python-oracledb.readthedocs.io/en/latest/user_guide/appendix_c.html#python-frameworks-sql-generators-and-orms

**Reply by jnre** [reply]
this is for thin mode. i do need thick mode for two-phase commit yes?

**Reply by CaselIT** [reply]
You can tell oracledb what mode to use with an env variable
